parser/src/token.rs

```diff
@@ -1,31 +1,71 @@
+//!All valid token types in the TokenStream
 use std::{fmt::Display, hash::Hash, num::ParseFloatError, str::FromStr};
-/// All the axepted tokens
+/// All the accepted tokens
 #[derive(Eq, PartialEq, Debug, Clone, Hash)]
 pub enum Token {
-    /// Any non other
+    /// Any other unknown chars not identified under
     Identifier(String),
+
+    /// A number with extra info
+    /// specifically matching [ `0` - `9` ] and `.`
     NumberLiteral(NumberLiteral),
+
     /// String representation:`\`
-    /// for command starts and seperating matrixes
     Backslash,
-    LeftCurlyBracket,  // {
-    RightCurlyBracket, // }
-    LeftBracket,       // [
-    RightBracket,      // ]
-    LeftParenthesis,   // (
-    RightParenthesis,  // )
-    Plus,              // +
-    Minus,             // -
-    Asterisk,          // *
-    Slash,             // /
-    Apostrophe,        // '
-    Underscore,        // _
-    Caret,             // ^
-    VerticalPipe,      // | and |
-    Comma,             // ,
-    Ampersand,         // &
-    Equals,            // =
-    EndOfContent,      // A special token that represents the end of content.
+
+    /// String representation:`{`
```

Comment:
I am not quite sure if this is more readable, it might be simpler to have it like it was before with a bigger comment on top explaining the smaller comments.

Reply:
It is more Ide redable but no not mutch else